src/coreclr/vm/prestub.cpp

```diff
@@ -1250,6 +1271,74 @@ namespace
         return true;
     }
 
+    void VerifyDeclarationSatifiesTargetConstraints(MethodDesc* declaration, MethodTable* targetType, MethodDesc* targetMethod)
```

Comment:
Add a test like the one from https://github.com/dotnet/runtime/pull/99468#discussion_r1524256899 ?